docs/proposed/psa-conditional-inclusion-c.md

```diff
@@ -80,9 +80,18 @@ If a feature is not requested for inclusion in the PSA Crypto configuration file
 
 #### Configuration symbols for key types
 
-For each constant or constructor macro of the form `PSA_KEY_TYPE_xxx`, the symbol **`PSA_WANT_KEY_TYPE_xxx`** indicates that support for this key type is desired.
+For most constant or constructor macros of the form `PSA_KEY_TYPE_xxx`, the symbol **`PSA_WANT_KEY_TYPE_xxx`** indicates that support for this key type is desired.
 
-For asymmetric cryptography, `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR` determines whether private-key operations are desired, and `PSA_WANT_KEY_TYPE_xxx_PUBLIC_KEY` determines whether public-key operations are desired. `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR` implicitly enables `PSA_WANT_KEY_TYPE_xxx_PUBLIC_KEY`: there is no way to only include private-key operations (which typically saves little code).
+As an exception, starting in Mbed TLS 3.5.0, for `KEY_PAIR` types (that is, private keys for asymmetric cryptography), the feature selection is more fine-grained, with an additional suffix:
+* `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_USE` enables support for operations with a key of that type (for enabled algorithms). This is automatically enabled if any key creation method (`IMPORT`, `GENERATE` or `DERIVE`) is enabled.
```

Comment:
> This is automatically enabled if any key creation method (`IMPORT`, `GENERATE` or `DERIVE`) is enabled.

Actually I think I'm enabling `USE` even for the `EXPORT` case. It makes sense in my opinion: also exporting requires some sort of support, i.e. `USE`. Wdyt?

Reply:
Yes, in my mind any `KEY_PAIR_xxx` should auto-enable `KEY_PAIR_USE`. I stole this text from @gilles-peskine-arm and failed to notice EXPORT was excluded.

Reply:
I think that when I wrote this, my thinking was: we auto-enable USE whenever there's any way to get a key into the system. EXPORT doesn't factor into that. But “whenever there's any way to get a key into the system” not exactly “if any creation method is enabled”: a system could have a built-in key (provisioned by proprietary means) and no way to create a new one.

I think what bothers me is that USE is not exactly about _using_ keys. It includes using keys, but it's also about knowing that at least some aspect of the key type is supported. For example, it's a promise that metadata function/macros (e.g. `xxx_MAX_SIZE`) take that key type into account. (Although I don't think we currently have any relevant macro that's unambiguously in the public API (either they're tied to export or they're tied to public-key support or they're tied to curve/group support or they're tied to algorithm support.)

So I'm now thinking that USE isn't a good name. @mpg [originally proposed](https://github.com/Mbed-TLS/mbedtls/issues/7439#issuecomment-1508109818) BASIC, which I think is a better name. I don't remember why we changed that name to USE. With the name BASIC, it's more intuitive that any form of support automatically enables BASIC.

Reply:
Yes, I tend to agree that the actual meaning of what we currently call `USE` is not just about using the key to perform operations, it's more generally "the PSA system knows about this key type".

I have nothing against switching to name that better reflects this, such as BASIC. I'm adding [a note about this](https://github.com/Mbed-TLS/mbedtls/issues/7439#issuecomment-1592673401) in the original discussion, and if nobody objects, I'll update this PR.

Reply:
It makes sense to me as well, so I'll update #7641 accordingly